mitosheet/src/mito/components/toolbar/GraphTabs/ChartDesignTabContents.tsx

```diff
@@ -0,0 +1,143 @@
+        analysisData: AnalysisData;
+        sheetDataArray: SheetData[];
+        selectedColumnsIds?: ColumnID[];
+        mitoContainerRef: React.RefObject<HTMLDivElement>;
+    }): JSX.Element => {
+    const currOpenTaskpane = props.uiState.currOpenTaskpane;
+    if (currOpenTaskpane.type !== TaskpaneType.GRAPH) {
+        throw new Error('ChartDesignTabContents should only be used when the graph taskpane is open.');
+    }
+    const openGraph = currOpenTaskpane.openGraph;
+    
+    // Get the step params and the step id of the currently open graph by getting the most recent step from this array
+    const stepSummaryList = props.analysisData.stepSummaryList;
+    const currGraphStep = stepSummaryList[stepSummaryList.length - 1];
+    const params = currGraphStep.params as GraphParamsBackend | undefined;
+
+    // When edits happen, just call the API directly w/ the graph id and the step id
+    const updateGraphParam = (update: RecursivePartial<GraphParamsBackend>): void => {
+        if (params === undefined) {
+            return;
+        }
+        props.mitoAPI.editGraph(
+            openGraph.graphID,
+            convertBackendtoFrontendGraphParams(updateObjectWithPartialObject(params, update)),
+            '100%',
+            '100%',
+            currGraphStep.step_id,
+            true
+        );
+    }
+
+    const graphDataArray = props.analysisData.graphDataArray;
+    const graphData = graphDataArray.find((graphData) => graphData.graph_id === openGraph.graphID);
+
+    // We append the correct export code for showing and for exporting to html
+    const [_copyShowGraphCode] = useCopyToClipboard(
+        (graphData?.graph_output?.graphGeneratedCode || '') + `\nfig.show(renderer="iframe")`
+    );
+    const [_copyExportHTMLGraphCode] = useCopyToClipboard(
+        (graphData?.graph_output?.graphGeneratedCode || '') + `\nfig.write_html("${graphData?.graph_tab_name}.html")`
+    );
+
+    const copyShowGraphCode = () => {
+        _copyShowGraphCode()
+
+        // Log that the user copied the graph code
+        void props.mitoAPI.log('copy_graph_code', {
+            'graph_type': params?.graph_creation?.graph_type
+        });
+    }
+    const copyExportHTMLGraphCode = () => {
+        _copyExportHTMLGraphCode()
+
+        // Log that the user copied the graph code
+        void props.mitoAPI.log('copy_export_html_graph_code', {
+            'graph_type': params?.graph_creation?.graph_type
+        });
+    }
+
+    return (<div className='mito-toolbar-bottom'>
+        <AddChartElementButton 
+            {...props}
+            params={params}
+            updateGraphParam={updateGraphParam}
+        />
+        <ChangeChartTypeButton
+            {...props}
+            params={params}
+            updateGraphParam={updateGraphParam}
+        />
+        
+        <div className='toolbar-vertical-line' />
+
+        {params === undefined ? <p> Loading... </p> : <GraphTypeConfigurations
```

Comment:
Are there other cases where params could be undefined?

Reply:
Not to my knowledge